File: src/c#/Creational/FactoryMethod/Util/EmployeeRandomizer.cs
using System;
namespace Util
{
    public class EmployeeRandomizer
    {

		static void Main()
		{ }

		private static readonly Random random = new Random();

		private static readonly String[] firstNames = { "Ali", "Ayse", "Bahar", "Bekir", "Bulent", "Can", "Cem", "Demet", "Elif",
			"Eylem", "Faruk", "Fatma", "Ganime", "Gulsum", "Haydar", "Halil", "Ismail", "Jale", "Kemal", "Leman",
			"Mehmet", "Mihrimah", "Murat", "Mustafa", "Nedim", "Nilufer", "Selim", "Selman", "Sevda", "S¸leyman",
			"Tarik", "Teoman", "Turgut", "Yeliz", "Zuhal" };

		private static readonly String[] departments = { "Production", "Sales", "Marketing", "Engineering" };

		public static int CreateId()
		{
			int randomInt = random.Next(100000);
			return randomInt;
		}

		public static String CreateName()
		{
			int randomInt = random.Next(35);
			return firstNames[randomInt];
		}

		public static String CreateDepartment()
		{
			int randomInt = random.Next(4);
			return departments[randomInt];
		}

		public static int CreateYear()
		{
			int randomInt = random.Next(20);
			return randomInt;
		}
	}
}

File: src/typescript/com/gokselkucuksahin/dp/creational-patterns/factory/employee-randomizer.ts
export class EmployeeRandomizer {
  private static firstNames = ["Ali", "Ayse", "Bahar", "Bekir", "Bulent", "Can", "Cem", "Demet", "Elif",
    "Eylem", "Faruk", "Fatma", "Ganime", "Gulsum", "Haydar", "Halil", "Ismail", "Jale", "Kemal", "Leman",
    "Mehmet", "Mihrimah", "Murat", "Mustafa", "Nedim", "Nilufer", "Selim", "Selman", "Sevda", "Süleyman",
    "Tarik", "Teoman", "Turgut", "Yeliz", "Zuhal"];

  private static departments = ["Production", "Sales", "Marketing", "Engineering"];

  private static getRandomInt(min: number, max: number): number {
    return ~~(Math.random() * (max - min) + min);
  }

  static createId(): number {
    return this.getRandomInt(0, 100_000);
  }

  static createName(): string {
    const randomIndex = this.getRandomInt(0, this.firstNames.length);
    return this.firstNames[randomIndex];
  }

  static createDepartment(): string {
    const randomIndex = this.getRandomInt(0, this.departments.length);
    return this.departments[randomIndex];
  }

  static createYear(): number {
    return this.getRandomInt(1, 20);
  }
}